### PR TITLE
Hide signatures

### LINKED ISSUE #3951: Hide forum signatures addon
This is kind of a no-brainer so I'm surprised it apparently hasn't been suggested yet :P

Simple addon which hides user signatures on forum posts when it's enabled. The "Full areas" addon expands signatures when active; there should be an opposite addon for those who don't care for signatures, want a break from banners and the like, or have any other reasons for hiding signatures!

Another option is to replace signatures with a (quite subtle) "expand signature" button that un-hides the sig when clicked (collapsing by default). This would make it so users don't have to disable the addon if they just want to have a quick look at one person's signature, for example.

### PR DESCRIPTION
**Pull request by FunctionalMetatable**
Resolves #3951

### Changes

Adds a hide signatures addon, which allows you to hide a signature

### Tests

Tested on Firefox

### CODE REVIEW

**Comment by hello-smile6** [comment]
Looks interesting! I wanna test this.

File: addons/hide-signatures/addon.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  "userscripts": [
+    {
+      "url": "userscript.js",
+      "matches": ["topics"],
+      "runAtComplete": true
+    }
+  ],
+  "settings": [
+    {
+      "name": "Hide signatures by default",
```
Comment (RedGuy12) [inline addons/hide-signatures/addon.json:19]: When would you not want to hide them by default if you have this addon?

File: addons/hide-signatures/userscript.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+export default async function ({ addon, global, console, msg }) {
+  // get signatures
+  const posts = document.querySelectorAll(".blockpost");
```
Comment (RedGuy12) [inline addons/hide-signatures/userscript.js:3]: WaitForElement

Comment (WorldLanguages) [reply addons/hide-signatures/userscript.js:3]: I'm not sure if there's consensus on whether waitForElement should be used for forum posts 🤔

File: addons/hide-signatures/addon.json (hunk 1 above)
Comment (FunctionalMetatable) [reply addons/hide-signatures/addon.json:19]: Idk

File: addons/hide-signatures/userscript.js (hunk 2 above)
Comment (FunctionalMetatable) [reply addons/hide-signatures/userscript.js:3]: The script loads and runs when the page is loaded. The posts will already be loaded as they are not added by JS. Addons which touches posts already use `document.querySelector`, not `addon.tab.waitForElement`

File: addons/hide-signatures/userscript.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  for (const post of posts) {
+    const actionsWrapper = post.querySelector(".postfootright");
+    const signature = post.querySelector(".postsignature");
```
Comment (WorldLanguages) [inline addons/hide-signatures/userscript.js:7]: `signature` might be `null`!

Comment (RedGuy12) [reply addons/hide-signatures/userscript.js:7]: ts be like

**Pull request merged**